src/ui/hooks/services/useStakingService.ts

```diff
@@ -38,43 +37,24 @@ export function useStakingService() {
   const { publicKeyNoCoord, address: btcAddress } = useBTCWallet();
   const { bech32Address } = useCosmosWallet();
   const logger = useLogger();
-  const { selectedProviderIds } = useFinalityProviderBsnState();
-
-  const calculateFeeAmount = ({
-    finalityProvider,
-    amount,
-    term,
-    feeRate,
-  }: Omit<FormFields, "feeAmount">) => {
-    logger.info("Calculating fee amount for EOI", {
-      finalityProvider: Array.isArray(finalityProvider)
-        ? finalityProvider.join(",")
-        : finalityProvider,
+
+  const calculateFeeAmount = useCallback(
+    ({
+      finalityProviders,
       amount,
       term,
       feeRate,
-    });
-    // Determine list of finality-provider public keys to be used when
-    // calculating the staking-fee. Priority order:
-    // 1. Providers selected in the multistaking flow (selectedProviderIds)
-    // 2. Providers coming from the form (can be a single pk or an array)
-    const fpList =
-      selectedProviderIds.length > 0
-        ? selectedProviderIds
-        : Array.isArray(finalityProvider)
-          ? finalityProvider
-          : [finalityProvider];
-
-    const eoiInput = {
-      finalityProviderPksNoCoordHex: fpList,
-      stakingAmountSat: btcToSatoshi(amount),
-      stakingTimelock: term,
-      feeRate: feeRate,
-    };
-    // Calculate the staking fee
-    const feeAmount = estimateStakingFee(eoiInput, feeRate);
-    return feeAmount;
-  };
+    }: Omit<FormFields, "feeAmount">) => {
+      const eoiInput = {
```

Comment:
Simplified the fee calculation given that we can have a single source of truth for the fps list